spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -171,7 +171,7 @@ public Table loadTable(Identifier ident, String version) throws NoSuchTableExcep
       SparkTable sparkTable = (SparkTable) table;
 
       Preconditions.checkArgument(
-          sparkTable.snapshotId() == null,
+          sparkTable.snapshotId() == null && sparkTable.branch() == null,
```

Comment:
I'm not sure whether we actually want to fix this as part of this PR or a separate PR, but in the Iceberg sync we briefly talked about making sure that `SELECT * from ns.table.branch_x VERSION AS OF ...` shouldn't be supported and should throw an error, which is what this check is doing

Reply:
Probably a separate PR.

Reply:
I've applied this and moved this to https://github.com/apache/iceberg/pull/9219